decidim-elections/app/views/decidim/elections/votes/_show_casted.html.erb

```diff
@@ -14,15 +14,25 @@
       </p>
     </div>
 
-    <%= link_to :elections, class: "button" do %>
-      <%= t("decidim.elections.votes.confirmed.back") %>
-    <% end %>
-
-    <% if params[:token] %>
+    <% if params[:token] && election.questionnaire.questions.any? %>
       <div>
-        <%= t("decidim.elections.votes.confirmed.experience") %>
+        <p class="text-small">
+          <%= t("decidim.elections.votes.confirmed.experience") %>
+        </p>
+      </div>
+      <div class="button">
         <%= link_to t("decidim.elections.votes.confirmed.feedback"), election_feedback_path(election, hash: params[:id], token: params[:token]) %>
       </div>
     <% end %>
+
+    <% if !election.questionnaire.questions.any? && current_user.nil? %>
```

Comment:
💬  `election.questionnaire.questions.any?` is called 3 times in this partial. What do you think about memoizing its result in the controller?

Reply:
Great suggestion! Thanks a lot! 👍